tests/rescoup/test_chopstep.cpp

```diff
@@ -165,7 +165,7 @@ struct SimulatorFixture
     OpmSimulatorTestCase simulator_wrapper;
     Simulator *simulator;
     Opm::Schedule& schedule;
-    Opm::ReservoirCouplingMaster rc_master;
+    Opm::ReservoirCouplingMaster<double> rc_master;
```

Comment:
Shouldn't there be a test with float, too?

Reply:
Thanks for the suggestion. I investigated adding float type testing, but discovered it would require extensive refactoring beyond this PR's scope.

The main blocker is that `AdaptiveTimeStepping` (where the `maybeChopSubStep()` will be used) is hardcoded to use `double` throughout. Templating it would be a major architectural change affecting core OPM components.

I've added a TODO comment in the test file documenting this limitation.

This should be tracked as technical debt for a future dedicated refactoring effort rather than expanding the current PR's scope.

Reply:
Well, my original intention was to enable gpu experimentation with full float simulators.